azurerm/internal/services/web/function_app_resource.go

```diff
@@ -611,6 +636,14 @@ func resourceFunctionAppRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("possible_outbound_ip_addresses", props.PossibleOutboundIPAddresses)
 		d.Set("client_affinity_enabled", props.ClientAffinityEnabled)
 		d.Set("custom_domain_verification_id", props.CustomDomainVerificationID)
+
+		if clientCertEnabled := props.ClientCertEnabled; clientCertEnabled != nil {
+			if *clientCertEnabled {
+				d.Set("client_cert_mode", props.ClientCertMode)
+			} else {
+				d.Set("client_cert_mode", "Ignore")
+			}
+		}
```

Comment:
using the changes above, this can become:

```suggestion
		clientCertMode := ""
		if props.ClientCertEnabled != nil {
			clientCertMode = *props.ClientCertMode
		}
		d.Set("client_cert_mode", clientCertMode)
```